lib/pkg.ml

```diff
@@ -436,6 +436,10 @@ let infer_name dir =
                   m "cannot determine name automatically: use `-p <name>`");
               exit 1 ) )
 
+let infer_main_pkg dir pkgs =
+  infer_name dir >>| fun main_pkg_name ->
+  List.find (fun pkg -> String.equal pkg.name main_pkg_name) pkgs
```

Comment:
I'm wondering if there might be projects where that strategy might fail. I think it's very unlikely. But even being very unlikely, do you think it might make sense to add a lint check that checks if the main package can be inferred (in a separate PR)?

Reply:
It makes a lot of sense indeed!